osconfig_tests/gcp_clients/gcp_clients.go

```diff
@@ -28,37 +29,57 @@ import (
 
 var (
 	storageClient  *storage.Client
+	computeClient  compute.Client
 	osconfigClient *osconfig.Client
 
 	populateClientOnce sync.Once
 )
 
-// PopulateClients creates singleton clients for various services
-func PopulateClients(ctx context.Context) {
+func populateClients(ctx context.Context) error {
+	var err error
 	populateClientOnce.Do(func() {
-		createOsConfigClient(ctx)
-		createStorageClient(ctx)
+		err = createComputeClient(ctx)
+		err = createOsConfigClient(ctx)
+		err = createStorageClient(ctx)
 	})
+	return err
```

Comment:
I'm not sure how the lifecycle of `err` is managed by go. If `populateClientOnce` has already executed and there was an error, won't err be nil the next time this is called and won't that cause problems for tests that check the err?